ghe-check-hotpatch.sh

```diff
@@ -89,11 +97,24 @@ check_log () {
 verify_running_image_tags () {
 # Get the running, and expected, hashes for all containerized service
 # and verify if they match.
+
+  # Cache JSON so we don't call sudo so many times
+  expected_json="$(sudo cat /data/user/docker/image/overlay2/repositories.json)"
   
   echo "Checking that all containers are running on their correct versions:"
-  for s in $(docker ps -q | xargs docker inspect --format='{{.Config.Image}}' | awk -F ':' '{print $1}' | sort -u); do
-    RUNNING_TAG=$(docker ps | grep "$s" | awk '{print $2}' | cut -d':' -f2 | sort -u)
-    EXPECTED_TAG=$(sudo cat /data/user/docker/image/overlay2/repositories.json | jq .Repositories | grep "$s:" | head -n1 | awk -F'\"' '{print $2}' | cut -d':' -f2)
+  while read id image; do
+    # $image format is <image>:<tag>
+    s="${image%%:*}"
+    RUNNING_TAG="${image#*:}"
+
+    # $image format is <image>:<tag>
+    image="$(jq -r '.Repositories."'"${s}"'" | keys[0]' <<<"${expected_json}")"
+    s2="${image%%:*}"
+    EXPECTED_TAG="${image#*:}"
+    if [ "${s2}" != "${s}" ]
+    then
+      echo "WARNING: Unexpected image name for '${s}' in expected tag list: ${s2}"
+    fi
```

Comment:
- Rather than retrieving the image, then extracting the (repository) name, the `docker ps` below will feed in the ID and image (complete with hash ("tag")) into the loop.
- We are able to then extract the name and hash using bash string manipulation alone.
- `jq` can then be used to directly retrieve the Repository in question by it's name from the expected list, and it's key which is its image (complete with hash ("tag")).
- We can then see if the 2 image (repository) names match - they should